llama_index/evaluation/semantic_similarity.py

```diff
@@ -0,0 +1,63 @@
+from typing import Any, Callable, Optional, Sequence
+from llama_index.evaluation.base import BaseEvaluator, EvaluationResult
+from llama_index.indices.service_context import ServiceContext
+from llama_index.embeddings.base import SimilarityMode, similarity
+
+
+class SemanticSimilarityEvaluator(BaseEvaluator):
+    """Embedding similarity evaluator.
+
+    Evaluate the quality of a question answering system by
+    comparing the similarity between embeddings of the generated answer
+    and the reference answer.
+
+    Args:
+        service_context (Optional[ServiceContext]): Service context.
+        similarity_threshold (float): Embedding similarity threshold for "passing".
+            Defaults to 0.8.
+    """
+
+    def __init__(
+        self,
+        service_context: Optional[ServiceContext] = None,
```

Comment:
Any reason to pass the entire service context rather than just the embed model?

Reply:
Good question, mostly to keep consistent with other evaluators. We should just change all of them to directly use embed_model or LLMs.